# HG changeset patch
# Parent 2741c6558064451a7ff52d66897fa3f3fe17ef81
# User Ben Kelly <ben@wanderview.com>
Resolve Cache::Match() with undefined if there is no match.

diff --git a/dom/cache/Cache.cpp b/dom/cache/Cache.cpp
--- a/dom/cache/Cache.cpp
+++ b/dom/cache/Cache.cpp
@@ -377,17 +377,17 @@ Cache::RecvMatchResponse(RequestId aRequ
   }
 
   if (NS_FAILED(aRv)) {
     promise->MaybeReject(aRv);
     return;
   }
 
   if (aResponse.type() == PCacheResponseOrVoid::Tvoid_t) {
-    promise->MaybeReject(NS_ERROR_DOM_NOT_FOUND_ERR);
+    promise->MaybeResolve(JS::UndefinedHandleValue);
     return;
   }
 
   nsRefPtr<Response> response = TypeUtils::ToResponse(mGlobal, aResponse,
                                                       aStreamControl);
   promise->MaybeResolve(response);
 }
 
